apollo-ios-codegen/Sources/ApolloCodegenLib/CodegenConfiguration/ApolloCodegenConfiguration.swift

```diff
@@ -283,7 +283,7 @@ public struct ApolloCodegenConfiguration: Codable, Equatable {
       case embeddedInTarget(name: String, accessModifier: AccessModifier = .internal)
       /// Generates a `Package.swift` file that is suitable for linking the generated schema types
       /// files to your project using Swift Package Manager.
-      case swiftPackageManager
+      case swiftPackageManager(version: ApolloPackageVersion = .default)
```

Comment:
When calling this in config, you're writing `moduleType: .swiftPackageManager(version:)` and there isn't great context on what "version" means here. I would assume I was setting the version number I want my schema module to have.

Possible other names for this parameter: 
- `apolloDependencyVersion`
- `apolloSDKVersion`
- `apolloVersion`

@calvincestari @BobaFetters, opinions?

Reply:
Hmmm yea i guess its not even necessarily a "version" its almost a dependency type, like which type do you want? Branch name? Commit hash? etc maybe `apolloDependencyType`?

Reply:
Went with `dependencyType` for now but can adjust further if need be.